commands/operator-sdk/cmd/add/api.go

```diff
@@ -162,7 +165,54 @@ func apiRun(cmd *cobra.Command, args []string) {
 		log.Fatalf("failed to create %v: %v", filePath, err)
 	}
 
-	// TODO: append rbac rule to deploy/rbac/role.yaml
+	// append rbac rule to deploy/role.yaml
+	roleFilePath := filepath.Join(fullProjectPath, "deploy", "role.yaml")
+	roleYAML, err := ioutil.ReadFile(roleFilePath)
+	if err != nil {
+		log.Fatalf("failed to read role manifest %v: %v", roleFilePath, err)
+	}
+	obj, _, err := cgoscheme.Codecs.UniversalDeserializer().Decode(roleYAML, nil, nil)
+	if err != nil {
+		log.Fatalf("failed to decode role manifest %v: %v", roleFilePath, err)
+	}
+	switch role := obj.(type) {
+	// TODO: use rbac/v1.
+	case *rbacv1beta1.Role:
+		pr := &rbacv1beta1.PolicyRule{}
+		apiGroupFound := false
+		for i := range role.Rules {
+			if role.Rules[i].APIGroups[0] == r.FullGroup {
+				apiGroupFound = true
+				pr = &role.Rules[i]
+				break
+			}
+		}
+		// check if the resource already exists
+		for _, resource := range pr.Resources {
+			if resource == r.Resource {
+				log.Fatalf("apiGroup (%v) resource (%v) already exists", pr.APIGroups[0], resource)
```

Comment:
No need to fail if the rule for that resource already exists. There's nothing more to do so just return.

Reply:
this case shouldn't happen in general. If it happens, we should at least print out an warning to indicate something went wrong.

Reply:
It's not an error. Could be that you add an api. Delete `pkg/apis/<group>/<version>` and re-add the api. The rule already exists so it shouldn't be an error. Nor a warning since it's not incorrect.

Probably just log.Printf:
```Go
log.Printf("deploy/role.yaml RBAC rules already up to date for resource (%v, %v), r.APIVersion, r.Kind)
```

Reply:
let's log it then.